js/src/setup-mc/setup-stepper/choose-audience/useAutoSetLocationCountriesEffect.js

```diff
@@ -2,43 +2,33 @@
  * External dependencies
  */
 import { useCallback, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
 
 /**
- * Internal dependencies
- */
-import useStoreCountry from '.~/hooks/useStoreCountry';
-import useGetCountries from '.~/hooks/useGetCountries';
-
-/**
- * Automatically set `location` to be `selected` and `countries` to be `[ storeCountryCode ]`
- * when `values.location === null && values.countries.length === 0` (i.e. when users first visit the page).
+ * Automatically set `location` to be `selected`
+ * and `countries` to be the value from Target Audience Suggestion API
+ * when `values.location === null && values.countries.length === 0`
+ * (i.e. when users first visit the page).
  *
- * If the `storeCountryCode` is not in the list of the MC supported countries, then `countries` would be set to empty array `[]`.
- *
- * This is done by calling `onChange` from `getInputProps`, simulating user's manual input action.
+ * The value setting is done by calling `onChange` from `getInputProps`,
+ * simulating user's manual input action.
  *
  * @param {Object} formProps formProps.
  */
 const useAutoSetLocationCountriesEffect = ( formProps ) => {
 	const { values, getInputProps } = formProps;
-	const { code: storeCountryCode } = useStoreCountry();
-	const { data } = useGetCountries();
-
-	const hasNoLocationCountries =
-		values.location === null && values.countries.length === 0;
-
-	const setLocationCountries = useCallback( () => {
-		if ( ! data ) {
-			return;
-		}
 
-		const countriesValue = data[ storeCountryCode ]
-			? [ storeCountryCode ]
-			: [];
+	const setLocationCountries = useCallback( async () => {
+		const data = await apiFetch( {
+			path: `wc/gla/mc/target_audience/suggestions`,
```

Comment:
💅 
What about using the `API_NAMESPACE` from `.~/data/constants` instead of `'wc/gla'`?

Reply:
I just quickly looked through #1146. This file will be removed, so perhaps we could do this in #1146 to avoid resolving conflict if we want to replace it.

Reply:
> What about using the `API_NAMESPACE` from `.~/data/constants` instead of `'wc/gla'`?

Good call! Done in [`bd3689b` (#1145)](https://github.com/woocommerce/google-listings-and-ads/pull/1145/commits/bd3689b2420d54893b32d45658dd912b4a47aa5b). 

Now I'm thinking I should find some time and change all `'wc/gla'` in the code base to `API_NAMESPACE`, so that we won't have copy-and-paste and then have this review comment again. 😆 

> perhaps we could do this in #1146 to avoid resolving conflict if we want to replace it.

I made the changes in this PR and also resolved the conflict in PR #1146. The idea is that in case PR #1146 is not desired or got reverted in the future, everything in this PR would still be working nicely.

Reply:
> Now I'm thinking I should find some time and change all `'wc/gla'` in the code base to `API_NAMESPACE`, so that we won't have copy-and-paste and then have this review comment again.

Upvote for this. That'd be great! 😃